numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -3105,22 +3105,24 @@ OBJECT_argmax(PyObject **ip, npy_intp n, npy_intp *max_ind,
     PyObject *mp = ip[0];
 
     *max_ind = 0;
-    i = 1;
-    while (i < n && mp == NULL) {
+    for (i = 1; i < n && mp == NULL; ++i) {
         mp = ip[i];
-        i++;
     }
-    for (; i < n; i++) {
-        ip++;
+    if (mp != NULL) {
```

Comment:
They way figured, if all entries are `NULL`, then at the end `i == n` and `mp == NULL`, but if the last entry is the first non-`NULL`, then at the end `i == n` but `mp != NULL`. It is probably a good idea to add test cases with both these situations.